smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/MemberShapeDecodeXMLGenerator.kt

```diff
@@ -43,51 +43,60 @@ abstract class MemberShapeDecodeXMLGenerator(
-                currContainerKey = ".member"
-                currContainerName = nextContainerName
-                containerUsedForDecoding = currContainerName
-                ifNilOrIfLetStatement = "if let $currContainerName = $currContainerName {"
-            } else {
-                writer.write("let $nextContainerName = $currContainerName.nestedContainerNonThrowable(keyedBy: CodingKeys.self, forKey: $currContainerKey)")
-                // currContainerKey is intentionally not updated. This container is only used to detect empty lists, not for decoding.
-                currContainerName = nextContainerName
-                containerUsedForDecoding = containerName
-                ifNilOrIfLetStatement = "if $currContainerName != nil {"
-            }
-
-            writer.openBlock(ifNilOrIfLetStatement, "} else {") {
-                val memberBuffer = "${memberName}Buffer"
-                val memberContainerName = "${memberName}Container"
-                val (memberTargetSymbol, memberTargetSymbolName) = nestedMemberTargetSymbolMapper(memberTarget)
-                writer.write("let $memberContainerName = try $containerUsedForDecoding.decodeIfPresent($memberTargetSymbolName.self, forKey: $currContainerKey)")
-                writer.write("var $memberBuffer:\$T = nil", memberTargetSymbol)
-                writer.openBlock("if let $memberContainerName = $memberContainerName {", "}") {
-                    writer.write("$memberBuffer = \$N()", memberTargetSymbol)
-                    renderListMemberItems(memberTarget, memberContainerName, memberBuffer)
-                }
-                renderAssigningDecodedMember(memberName, memberBuffer)
+        writer.openBlock(ifNilOrIfLetStatement, "} else {") {
+            val memberBuffer = "${memberName}Buffer"
+            val memberContainerName = "${memberName}Container"
+            val (memberTargetSymbol, memberTargetSymbolName) = nestedMemberTargetSymbolMapper(memberTarget)
+            writer.write("let $memberContainerName = try $containerUsedForDecoding.decodeIfPresent($memberTargetSymbolName.self, forKey: $currContainerKey)")
+            writer.write("var $memberBuffer:\$T = nil", memberTargetSymbol)
+            writer.openBlock("if let $memberContainerName = $memberContainerName {", "}") {
+                writer.write("$memberBuffer = \$N()", memberTargetSymbol)
+                renderListMemberItems(memberTarget, memberContainerName, memberBuffer)
             }
-            writer.indent()
-            renderAssigningSymbol(memberName, "[]")
-            writer.dedent().write("}")
+            renderAssigningDecodedMember(memberName, memberBuffer)
         }
         writer.indent()
-        renderAssigningNil(memberName)
+        renderAssigningSymbol(memberName, "[]")
```

Comment:
What was the need to assign [] vs nil? I'm surprised this didn't break a bunch of stuff?